js/ui/viewSelector.js

```diff
@@ -199,9 +314,99 @@ const ViewsDisplay = new Lang.Class({
     Name: 'ViewsDisplay',
 
     _init: function() {
+        this._enterSearchTimeoutId = 0;
+        this._localSearchMetricTimeoutId = 0;
+
         this._appDisplay = new AppDisplay.AppDisplay()
 
-        this.actor = new ViewsDisplayContainer(this._appDisplay);
+        this._searchResults = new Search.SearchResults();
+        this._searchResults.connect('search-progress-updated', Lang.bind(this, this._updateSpinner));
+
+        // Since the entry isn't inside the results container we install this
+        // dummy widget as the last results container child so that we can
+        // include the entry in the keynav tab path
+        this._focusTrap = new FocusTrap({ can_focus: true });
+        this._focusTrap.connect('key-focus-in', Lang.bind(this, function() {
+            this.entry.grab_key_focus();
+        }));
+        this._searchResults.actor.add_actor(this._focusTrap);
+
+        global.focus_manager.add_group(this._searchResults.actor);
+
+        this.entry = new ShellEntry.OverviewEntry();
+        this.entry.connect('search-activated', Lang.bind(this, this._onSearchActivated));
+        this.entry.connect('search-active-changed', Lang.bind(this, this._onSearchActiveChanged));
+        this.entry.connect('search-navigate-focus', Lang.bind(this, this._onSearchNavigateFocus));
+        this.entry.connect('search-terms-changed', Lang.bind(this, this._onSearchTermsChanged));
+
+        this.entry.clutter_text.connect('key-focus-in', Lang.bind(this, function() {
+            this._searchResults.highlightDefault(true);
+        }));
+        this.entry.clutter_text.connect('key-focus-out', Lang.bind(this, function() {
+            this._searchResults.highlightDefault(false);
+        }));
+
+        // Clicking on any empty area should exit search and get back to the desktop.
+        let clickAction = new Clutter.ClickAction();
+        clickAction.connect('clicked', Lang.bind(this, this._resetSearch));
+        Main.overview.addAction(clickAction, false);
+        this._searchResults.actor.bind_property('mapped', clickAction, 'enabled', GObject.BindingFlags.SYNC_CREATE);
+
+        this.actor = new ViewsDisplayContainer(this.entry, this._appDisplay, this._searchResults);
+    },
+
+    _updateSpinner: function() {
+        this.entry.setSpinning(this._searchResults.searchInProgress);
+    },
+
+    _enterLocalSearch: function() {
```

Comment:
Yes, this also ended up getting addressed on a later commit.  Now, I see...  This commit and the next one are a bit mangled in the history.  I'll see if I can clean that up before merging.